### DISCUSSION TITLE
关于kraken.methodChannel.setMethodCallHandler使用方式

### DISCUSSION THREAD

**Discussion by zhuyashan** [openkraken/kraken#197] [Q&A] [answered]
在js端， 在constructor方法里面添加了kraken.methodChannel.setMethodCallHandler方法。在flutter端调用javaScriptChannel.invokeMethod。js端处理函数被没有被调用。kraken.methodChannel.setMethodCallHandler该如何使用，有没有demo供参考。

**Comment by andycall** [comment, marked as answer]
https://openkraken.com/guide/advanced/communicate-with-native

**Reply by zhuyashan** [reply]
这个文档我看过了。https://github.com/openkraken/kraken/issues/5 我用这种插件化的方案是没问题的。

**Reply by andycall** [reply]
那你是不是有可能在还没有执行 `kraken.methodChannel.setMethodCallHandler` 的时候，flutter 那边的 javaScriptChannel.invokeMethod 就调用了

**Reply by zhuyashan** [reply]
不是。等有空我再看看，还不行的话我写一个demo再来请教。

**Reply by andycall** [reply]
这是个 bug，在这个 PR 里面修复了 

https://github.com/openkraken/kraken/pull/289
